CoreUtilities/src/au/gov/asd/tac/constellation/utilities/query/QueryEvaluator.java

```diff
@@ -15,110 +15,227 @@
  *
  * @author aldebaran30701
  */
 public class QueryEvaluator {
+
+    private static final String PARENTHESES_REGEX = "(?<!(?<![^\\\\]\\\\(?:\\\\"
+            + "{2}){0,10})\\\\)\\)|(?<!(?<![^\\\\]\\\\(?:\\\\{2}){0,10})\\\\)\\(";
     
-    private static final String SPLIT_REGEX = "(?<!(?<![^\\\\]\\\\(?:\\\\{2}){0,10})\\\\)\\)|(?<!(?<![^\\\\]\\\\(?:\\\\{2}){0,10})\\\\)\\(";
-    
-    private enum Operator {
-        AND(1),
-        OR(2);
+    private QueryEvaluator() {
+    }
 
-        final int precedence;
+    /**
+     * Tokeniser takes input from the user and breaks it into terms such as
+     * <ul>
+     * <li>color == #ffffff</li>
+     * <li>selected == true</li>
+     * <li>||</li>
+     * <li>&&</li>
+     * <li>(</li>
+     * <li>)</li>
+     * </ul>
+     * <p>
+     * A backslash can be used to escape one of the terms. eg.
+     * <ul>
+     * <li>color == \(#ffffff\) - Input by user</li>
+     * <li>color == (#ffffff) - Evaluated as</li>
+     * </ul>
+     * <p>
+     * iterate all chars in input - check if moreToEscape if (, add term to list
+     * of tokens if |, and previousChar was | then add || to list of tokens if
+     * &, and previousChar was & then add && to list of tokens if ), add term to
+     * list of tokens else, an escaped character or normal input so add to
+     * currentString
+     * <p>
+     * @param input the query as written by the user to tokenise.
+     * @return List<String> stringTokens is the items within the input string,
+     * stored as a list.
+     */
+    public static List<String> tokeniser(final String input) {
+        final List<String> stringTokens = new ArrayList<>();
+        String currentString = "";
+        char prevChar = Character.UNASSIGNED;
 
-        private Operator(final int precedence) {
-            this.precedence = precedence;
+        if (StringUtils.isBlank(input)) {
+            return Collections.emptyList();
         }
-    }
-    
-    private static final Map<String, Operator> OPERATORS = Map.ofEntries(Map.entry("&&", Operator.AND),Map.entry("||", Operator.OR));
 
-    private static boolean isHigerPrecedence(final String operator, final String subOperator) {
-        return (OPERATORS.containsKey(subOperator) 
-                && OPERATORS.get(subOperator).precedence >= OPERATORS.get(operator).precedence);
+        for (final char c : input.toCharArray()) {
+            if (c == '(' && prevChar != '\\') {
+                if (StringUtils.isBlank(currentString)) {
+                    currentString = "(";
+                }
+                currentString = StringUtils.trim(currentString);
+                stringTokens.add(currentString);
+                currentString = "";
+            } else if (c == '|' && prevChar == '|') {
+                currentString = currentString.substring(0, currentString.lastIndexOf(prevChar));
+                if (!StringUtils.isBlank(currentString)) {
```

Comment:
There is actually a function `StringUtils.isNotBlank`. I'd use that instead (and do this for the other instances of this in the file)